rails/spec/controllers/admin/cohorts_controller_spec.rb

```diff
@@ -0,0 +1,225 @@
+    @cohort_3 = FactoryBot.create(:admin_cohort, name: 'cohort 3', project: project_3)
+  end
+
+  let(:project_1) { FactoryBot.create(:admin_project, name: 'project_1') }
+  let(:project_2) { FactoryBot.create(:admin_project, name: 'project_2') }
+  let(:project_3) { FactoryBot.create(:admin_project, name: 'project_3') }
+
+  let(:admin_user) { FactoryBot.generate(:admin_user) }
+  let(:user) { FactoryBot.create(:user) }
+
+  describe 'A user not affiliated with a project' do
+    describe 'GET index' do
+      it 'wont see any cohorts' do
+        get :index
+        expect(assigns[:admin_cohorts]).to eq([])
+      end
+    end
+
+    describe 'Show' do
+      it 'wont let them see any cohorts' do
+        [@cohort_1,@cohort_2,@cohort_3].each do |cohort|
+          get :show, id: cohort.id
+          # Redirect, and show error when not allowed:
+          expect(response).to have_http_status(:redirect)
+          expect(request.flash[:alert]).to match(RegexForAuthFailShow)
+        end
+      end
+    end
+
+    describe 'New' do
+      it 'it wont let them create a new cohort' do
+        get :new
+        # Redirect, and show error when not allowed:
+        expect(response).to have_http_status(:redirect)
+        expect(request.flash[:alert]).to match(RegexForAuthFailNew)
+      end
+    end
+
+    describe 'create' do
+      it 'it wont let them create a new cohort' do
+        put :create
+        # Redirect, and show error when not allowed:
+        expect(response).to have_http_status(:redirect)
+        expect(request.flash[:alert]).to match(RegexForAuthFailNew)
+      end
+    end
+
+    describe 'update' do
+      it 'it wont let them update an existing cohort' do
+        put :update, id:@cohort_1.id
+        # Redirect, and show error when not allowed:
+        expect(response).to have_http_status(:redirect)
+        expect(request.flash[:alert]).to match(RegexForAuthFailModify)
+      end
+    end
+  end
+
+  describe 'A user who is an admin for project 2' do
+    before(:each) do
+      user.add_role_for_project('admin', project_2)
+    end
+
+    describe 'GET index' do
+      it 'can only see cohort 2' do
+        get :index
+        expect(assigns[:admin_cohorts]).not_to include(@cohort_1)
+        expect(assigns[:admin_cohorts]).to include(@cohort_2)
+        expect(assigns[:admin_cohorts]).not_to include(@cohort_3)
+      end
+    end
+  end
+
+  describe 'A user who is an admin for project 1' do
+    before(:each) do
+      user.add_role_for_project('admin', project_1)
+    end
+
+    describe 'GET index' do
+      it 'can only see cohort 1' do
+        get :index
+        expect(assigns[:admin_cohorts]).to include(@cohort_1)
+        expect(assigns[:admin_cohorts]).not_to include(@cohort_2)
+        expect(assigns[:admin_cohorts]).not_to include(@cohort_3)
+      end
+    end
+
+    describe 'Show' do
+      describe 'Cohort 1 (user IS a memeber)' do
```

Comment:
Since cohorts can have teacher members. Using the word member here is confusing. If 'member' was changed 'admin of project', that would be more clear.  Same for below.